src/main/java/io/jenkins/plugins/coverage/adapter/LLVMCovReportAdapter.java

```diff
@@ -0,0 +1,276 @@
+package io.jenkins.plugins.coverage.adapter;
+
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import hudson.Extension;
+import io.jenkins.plugins.coverage.adapter.converter.JSONDocumentConverter;
+import io.jenkins.plugins.coverage.adapter.parser.LLVMCoverageParser;
+import io.jenkins.plugins.coverage.exception.CoverageException;
+import io.jenkins.plugins.coverage.targets.CoverageResult;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class LLVMCovReportAdapter extends JSONCoverageReportAdapter {
+
+    /**
+     * @param path Ant-style path of report files.
+     */
+    @DataBoundConstructor
+    public LLVMCovReportAdapter(String path) {
+        super(path);
+    }
+
+    @Override
+    protected JSONDocumentConverter getConverter() {
+        return new LLVMCovDocumentConverter();
+    }
+
+    @CheckForNull
+    @Override
+    protected CoverageResult parseToResult(Document document, String reportName) {
+        return new LLVMCoverageParser(reportName).parse(document);
+    }
+
+
+    @Symbol("llvm")
+    @Extension
+    public static class LLVMCovReportAdapterDescriptor extends CoverageReportAdapterDescriptor<CoverageReportAdapter> {
+
+        public LLVMCovReportAdapterDescriptor() {
+            super(LLVMCovReportAdapter.class);
+        }
+
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return Messages.LLVMCovReportAdapter_displayName();
+        }
+    }
+
+
+    public static class LLVMCovDocumentConverter extends JSONDocumentConverter {
+
+        @Override
+        protected Document convert(JSONObject report, Document document) throws CoverageException {
+            // only support 2.0.0 version now
+            if (!report.getString("version").equals("2.0.0") || !report.getString("type").equals("llvm.coverage.json.export")) {
+                throw new CoverageException("Unsupported Json file");
```

Comment:
Could we separate these two checks, and provide more detailed exceptions, e.g. `"Unsupported Json file - type must be "llvm.coverage.json.export"`? This would help the user better understand what's wrong

Reply:
Yeah